### PR TITLE
Add ability to query balance of reward accounts

### LINKED ISSUE #17: Add support for querying reward accounts
**Task description**

The block endpoint currently emits a phony transaction that includes operations for tokens moving to and from the synthetic accounts `baking_reward_account` and `finalization_reward_account`.

When running the official CLI checker against the implementation, these operations cause the tool to do balance lookups on these accounts. As this currently isn't supported, the checker fails.

Adding this ability will hopefully allow us to successfully verify the Data API using the official tool.

### PR DESCRIPTION
**Pull request by bisgardo**
Resolves #17.

### CODE REVIEW

**Comment by bisgardo** [comment]
Currently running `rosetta-cli check:data` full verification (with default configuration) against testnet service node from testnet gateway.

File: src/api/query.rs [hunk 1]
```diff
@@ -90,13 +109,28 @@ pub fn block_hash_from_string(hash: &str) -> ApiResult<BlockHash> {
     })
 }
 
-pub fn account_address_from_identifier(id: &AccountIdentifier) -> ApiResult<AccountAddress> {
+pub enum Address {
```
Comment (shjortConcordium) [inline src/api/query.rs:112]: Feels a bit misplaced, but don't know where else you would put it.

File: src/api/query.rs [hunk 2]
```diff
@@ -90,13 +109,28 @@ pub fn block_hash_from_string(hash: &str) -> ApiResult<BlockHash> {
...
+    if addr == ACCOUNT_BAKING_REWARD {
+        return Ok(Address::BakingRewardAccount);
+    }
+    if addr == ACCOUNT_FINALIZATION_REWARD {
+        return Ok(Address::FinalizationRewardAccount);
+    }
+    match AccountAddress::from_str(addr) {
+        Ok(a) => Ok(Address::Account(a)),
+        Err(_) => Err(ApiError::InvalidAccountAddress(addr.to_string())),
+    }
```
Comment (shjortConcordium) [inline src/api/query.rs:135]:
```suggestion
    match addr {
        ACCOUNT_BAKING_REWARD => Ok(Address::BakingRewardAccount),
        ACCOUNT_FINALIZATION_REWARD => Ok(Address::FinalizationRewardAccount),
        _ => {
            match AccountAddress::from_str(addr) {
                Ok(a) => Ok(Address::Account(a)),
                Err(_) => Err(ApiError::InvalidAccountAddress(addr.to_string())),
            }
        }
    }
```
is maybe a bit nicer? :shrug: 
I tried to only have a single match, but it doesn't seem possible without doing stupid stuff

**Review by shjortConcordium — COMMENTED** [review]
LGTM, though have only looked at it. Surprised that we only used accountInfo to get the balance. 
(I can test it tomorrow if necessary)

File: src/api/query.rs (hunk 1 above)
Comment (abizjak) [inline src/api/query.rs:112]: A comment explaining the semantics of the type, and why it exists, would be helpful..

**Review by abizjak — APPROVED** [review]

**Comment by shjortConcordium** [comment]
> What do you mean that "we only used accountInfo to get the balance"?

Just an inane comment: That we could replace `query_account_info` with a `query_account_balance`, because it wasn't used for anything else.

File: src/api/account.rs [hunk 3]
```diff
@@ -32,16 +32,16 @@ impl AccountApi {
...
-            .query_account_info(req.block_identifier, req.account_identifier.deref())
+            .query_account_balance(req.block_identifier, req.account_identifier.deref())
             .await?;
         Ok(AccountBalanceResponse::new(
             BlockIdentifier::new(
                 block_info.block_height.height as i64,
                 block_info.block_hash.to_string(),
             ),
-            vec![amount_from_uccd(account_info.account_amount.microgtu as i64)],
+            vec![amount_from_uccd(amount.microgtu as i64)],
```
[the remark this answers is not in the document]
Comment (bisgardo) [reply src/api/account.rs:44]: Fixed in separate PR: https://github.com/Concordium/concordium-rosetta/pull/21.

File: src/api/query.rs (hunk 1 above)
Comment (bisgardo) [reply src/api/query.rs:114]: Agree and agree :)

Comment (bisgardo) [reply src/api/query.rs:114]: Done.

File: src/api/query.rs (hunk 2 above)
Comment (bisgardo) [reply src/api/query.rs]: Strange, tried that at first but were certain that it interpreted the constants as an ordinary "anything" match. Applied the change manually.

**Pull request merged**